sdl_android/src/main/java/com/smartdevicelink/api/audio/BaseAudioDecoder.java

```diff
@@ -0,0 +1,226 @@
+package com.smartdevicelink.api.audio;
+
+import android.content.Context;
+import android.media.AudioFormat;
+import android.media.MediaCodec;
+import android.media.MediaExtractor;
+import android.media.MediaFormat;
+import android.net.Uri;
+import android.os.Build;
+import android.support.annotation.NonNull;
+import android.support.annotation.RequiresApi;
+import android.util.Log;
+import com.smartdevicelink.api.audio.AudioStreamManager.SampleType;
+
+import java.nio.ByteBuffer;
+import java.nio.ByteOrder;
+
+@SuppressWarnings("WeakerAccess")
+@RequiresApi(api = Build.VERSION_CODES.JELLY_BEAN)
+public abstract class BaseAudioDecoder {
+    private static final String TAG = AudioDecoder.class.getSimpleName();
+
+    protected int targetSampleRate;
+    protected @SampleType int targetSampleType;
+
+    private int outputChannelCount;
+    private int outputSampleRate;
+    private @SampleType int outputSampleType;
+
+    private double lastOutputSample = 0;
+
+    private double lastOutputPresentationTimeUs = 0;
+    private double lastTargetPresentationTimeUs = 0;
+
+    protected MediaExtractor extractor;
+    protected MediaCodec decoder;
+
+    protected Uri audioSource;
+    protected Context context;
+    protected AudioDecoderListener listener;
+
+    public BaseAudioDecoder(Uri audioSource, Context context, int sampleRate, @SampleType int sampleType, AudioDecoderListener listener) {
+        this.audioSource = audioSource;
+        this.context = context;
+        this.listener = listener;
+
+        targetSampleRate = sampleRate;
+        targetSampleType = sampleType;

```

Comment:
See previous comment. Javadoc should be added to address possible values as well as some error checking at this point to ensure an error does not persist downstream (eg `SampleBuffer`).

Reply:
Done